src/lib/corresponding-file.ts

```diff
@@ -0,0 +1,25 @@
+import * as path from 'path'
+import { getExtension, stripExtension } from './path-utils'
+
+function escapeRegex(string: string) {
+  return string.replace(/[/\-\\^$*+?.()|[\]{}]/g, '\\$&')
+}
+
+export function getCorrespondingStyleFile(
+  inputPath: string,
+  matchPattern: string,
+  outputFile: string,
+) {
+  const basePath = path.dirname(inputPath)
+  const filename = stripExtension(path.basename(inputPath))
+  const extension = getExtension(path.basename(inputPath))
+  const regex = `^${escapeRegex(matchPattern).replace('\\*', '(.*)')}$`
+  if (!filename.match(regex)) {
+    return null
+  }
+
+  const outputFilename = filename.replace(new RegExp(regex), outputFile)
+  const outputExtension = extension.match(/tsx?/) ? 'ts' : 'js'
```

Comment:
We automatically determine the extension type so the extension works with both JS and TS out of the box.